app/main.py

```diff
@@ -1,3 +1,16 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    cashed_data = {}
+
+    def inner_function(*args) -> Any:
+        if args in cashed_data:
+            print("Getting from cache")
+            return cashed_data[args]
+        else:
+            print("Calculating new result")
+            func_returned_value = func(*args)
+            cashed_data[args] = func_returned_value
+            return func_returned_value
```

Comment:
u can use a single return statement out of the if/else statement instead of two inside it

Reply:
I've changed something. Is it better now? =)